Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
You should not manually operate with the`process_table_` variable. In order to add object to process handler you should just set some non-zero frequency with the `SetFreq` function. But be carefull - you should not call this function in class constructor. If it is needed to start processing of an object from the beginning (like projectile) then you should call `SetFreq` from `AfterWorldCreation` (do not forget to call base `AfterWorldCreation` function, you can see example in `Lobby.cpp` file). `AfterWorldCreation` is called by game when world is fully created, it is like the second constructor. It should be used when it is needed to interact with outer world on creation.